hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/expunge/JpaResourceExpungeService.java

```
@@ -315,16 +317,22 @@ private void expungeCurrentVersionOfResource(
 
 		deleteAllSearchParams(JpaPid.fromId(resource.getResourceId()));
 
-		if (resource.isHasTags()) {
-			myResourceTagDao.deleteByResourceId(resource.getId());
-		}
+		try {
+			if (resource.isHasTags()) {
+				myResourceTagDao.deleteByResourceId(resource.getId());
+			}
 
-		if (resource.getForcedId() != null) {
-			ForcedId forcedId = resource.getForcedId();
-			myForcedIdDao.deleteByPid(forcedId.getId());
-		}
+			if (resource.getForcedId() != null) {
+				ForcedId forcedId = resource.getForcedId();
+				myForcedIdDao.deleteByPid(forcedId.getId());
+			}
 
-		myResourceTableDao.deleteByPid(resource.getId());
+			myResourceTableDao.deleteByPid(resource.getId());
+		} catch (Exception e) {
+			throw new PreconditionFailedException(Msg.code(2415)
```

If you really want to catch all exceptions, then we should log the stack trace in case something weird is going on.  Just appending the exception (with `+ e`) will only log the error message, not the whole stack trace.  The stack trace is important debugging information.

---

Got it. I'll keep that in mind